app/scripts/shell/dimSearchFilter.directive.js

```diff
@@ -94,6 +94,10 @@
 
     vm.search = dimSearchService;
 
+    $scope.$watch('vm.search.query', function() {
```

Comment:
It is because I'm allowing the query to be manipulated in ways other than through the field.